src/main/java/org/folio/circulation/domain/validation/ServicePointPickupLocationValidator.java

```diff
@@ -37,8 +38,15 @@
     }
 
     if(request.getPickupServicePointId() == null) {
-      log.info("No pickup service point specified for request");
-      return succeeded(requestAndRelatedRecords);
+      if(request.getFulfilmentPreference() == RequestFulfilmentPreference.HOLD_SHELF) {
+        log.info("Hold Shelf Fulfilment Requests require a Pickup Service Point");
+        return failed(ValidationErrorFailure.failure(
```

Comment:
Personally, I prefer to statically import the `failure` method, as I believe it improves the readability. What do you think 
```suggestion
        return failed(failure(
```

Reply:
Maybe we should use `failedResult` more often: https://github.com/folio-org/mod-circulation/blob/dda4244b1a7d42d2ce3758bb4595939c0855196e/src/main/java/org/folio/circulation/support/ValidationErrorFailure.java#L22-L29

I'm all for static imports, even in this case, however I think `failed(failure(` looks a bit... weird. 😄

Reply:
Agreed. Whilst reviewing this, I've been thinking that the `failedResult` method needs tidying up (the parameters need renaming at least).

And I'm wondering if it might be better to rename the method to `fail`, to improve how this reads.

I think both of those changes are out of scope of this change, however would welcome both of your thoughts?

Reply:
I was leveraging Eclipse IDE fanciness to find that constant and increase the Request.getFulfilmentPreference()'s visibility. I didn't notice that I ended up with a static import. I will update the PR.

Reply:
Hmm, I copied and pasted the 'failed failure' format from further down in the file.

I wouldn't want to convert just this new instance to the shorter version, and refactoring the whole file seems like it shouldn't happen as part of a separate Issue PR.